### PR TITLE
Using BaseDataHandler

### PR DESCRIPTION
**Pull request by baulus**

### CODE REVIEW

**Comment by delliott90** [comment]
Looks good. Does anything need to be updated in the developer guide? https://github.com/IBM/cp4s-car-reference-connector/blob/develop/guide-build-connectors.md

**Comment by baulus** [comment]
> Looks good. Does anything need to be updated in the developer guide? https://github.com/IBM/cp4s-car-reference-connector/blob/develop/guide-build-connectors.md

No changes required in https://github.com/IBM/cp4s-car-reference-connector/blob/develop/guide-build-connectors.md#connectordata_handlerpy

File: connector/data_handler.py [hunk 1]
```diff
@@ -32,16 +33,21 @@ def get_report_time():
...
-        # create source entry and it is compuslory for each imports API call
-        self.source = {'_key': context().args.source, 'name': context().args.server, 'description': 'Reference Asset server'}
-        self.report = {'_key': str(now), 'timestamp' : now, 'type': 'Reference Asset server', 'description': 'Reference Asset server'}
+
+    def create_source_report_object(self):
+        if not (self.source and self.report and self.source_report):
+            # create source, report and source_report entry and it is compuslory for each imports API call
+            self.source = {'_key': context().args.source, 'name': context().args.server, 'description': 'Reference Asset server'}
+            self.report = {'_key': str(self.timestamp), 'timestamp' : self.timestamp, 'type': 'Reference Asset server', 'description': 'Reference Asset server'}
+            self.source_report = [{'active': True, '_from': 'source/' + self.source['_key'], '_to': 'report/' + self.report['_key'], 'timestamp': self.report['timestamp']}]
```
Comment (omkar-g) [inline connector/data_handler.py]: There is no source_report anymore

Comment (baulus) [reply connector/data_handler.py]: removed

**Review by omkar-g — APPROVED** [review]

**Pull request merged**